R/correctionsataglance-data.R

```diff
@@ -342,13 +344,15 @@ findTextLocations <- function(dataIn, isDateData = FALSE, ...){
   return(list(x = x, y = y))
 }
 
-isTextLong <- function(labelText, dateLim, startD, endD){
-  early <- which(startD < dateLim[1])
-  late <- which(endD > dateLim[2])
-  startD[early] <- dateLim[1]
-  endD[late] <- dateLim[2]
+isTextLong <- function(labelText, dateLim = NULL, startD, endD, totalDays = NULL){
+  if(is.null(totalDays)){
+    early <- which(startD < dateLim[1])
+    late <- which(endD > dateLim[2])
+    startD[early] <- dateLim[1]
+    endD[late] <- dateLim[2]
+    totalDays <- difftime(dateLim[2], dateLim[1], units="days")
+  } 
   
-  totalDays <- difftime(dateLim[2], dateLim[1], units="days")
   widthOfChar <- (1/365)*totalDays*3 #each character will be 1/365 * num days in the range * buffer
   widthOfLabel <- nchar(labelText)*widthOfChar
   widthOfRect <- difftime(endD, startD, units="days")
```

Comment:
So, need to figure out how to pass it the start and end dates for that specific section of the rectangle.. I think. Rather than the entire range for the request.